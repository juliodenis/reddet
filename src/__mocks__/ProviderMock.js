import React from "react";
import { createStore } from "redux";
import { Router } from "react-router-dom";
import { Provider } from "react-redux";
import { createBrowserHistory } from "history";
import reducers from "../redux/reducers";

const initialState = {
  posts: [],
  users: [],
};

const store = createStore(reducers, initialState);
const history = createBrowserHistory();

const ProviderMock = (props) => (
  <Provider store={store}>
    <Router history={history}>{props.children}</Router>
  </Provider>
);

export default ProviderMock;
