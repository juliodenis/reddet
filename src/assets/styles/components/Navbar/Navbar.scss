@import "../../Breackpoints.scss";
@import '../../partials/shadows.scss';
@import '../../partials/colors.scss';

.navbar {
    display: flex;
    justify-content: space-around;
    align-items: center;
    height: 80px;
    box-shadow: $shadow-normal;
    background-color: $color-primary;
    &__logo {
        a {

            text-transform: uppercase;
            color: $color-white;
            font-size: 28px;
            
        }

    }
    &__profile {
        .navbar__profile--settings {
            position: relative;
                .navbar__profile--settings-dropdown {
                    position: relative;
                    background-color: transparent;
                    border: none;
                    cursor: pointer;
                    display: flex;
                    flex-direction: row;
                    align-items: center;
                    justify-content: center;
                    font-size: 20px;
                    img {
                        width: 50px;
                        height: 50px;
                        border-radius: 50%;
                        object-fit: cover;
                    }
                    svg {
                        color: $color-background;
                    }
                }
        }
        .navbar__profile--settings:hover {
            .profile-dropdown-content {
                display: block;
            }
            a {
                background-color: $color-black-10;
                border-radius: 5px;
            }
        }
        .profile-dropdown-content {
            display: none;
            position: absolute;
            right: 0;
            background-color: $color-background;
            border: 1px solid $color-black-10;
            border-radius: 12px;
            box-shadow: $shadow-small-green;
            min-width: 180px;
            z-index: 1;
            a {
                color: $color-black-60;
                padding: 12px 16px;
                text-decoration: none;
                display: flex;
                justify-content: space-evenly;
                align-items: center;
                transition: .3s;
            }
            a:hover {
                background-color: $color-black-10;
                color: $color-primary;
            }
            
          }
        
    }
    
}