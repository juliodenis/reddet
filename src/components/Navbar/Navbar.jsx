import React, { Fragment, useState, useEffect } from "react";
import { connect, useDispatch } from "react-redux";
import { Link } from "react-router-dom";
import { VscEdit } from "react-icons/vsc";
import { CgLogOut } from "react-icons/cg";
import defaultImage from "../../assets/static/defaultUser.png";

import { startLogoutUser } from "../../redux/actions/auths";
import { startGetUser } from "../../redux/actions/users";

import { MdArrowDropDown } from "react-icons/md";

import "../../assets/styles/components/Navbar/Navbar.scss";

const Navbar = (props) => {
  const { image } = props.auth.users;
  const dispatch = useDispatch();

  useEffect(() => {
    const user = JSON.parse(localStorage.getItem("auth"));
    dispatch(startGetUser(user));
  }, []);

  const handleClick = (event) => {
    localStorage.removeItem("auth");
  };
  const Menu = () => {
    return (
      <div className="navbar__profile--settings">
        <a className="navbar__profile--settings-dropdown">
          <img src={image ? image : defaultImage} alt="user profile" />
          <MdArrowDropDown />
        </a>
        <div className="profile-dropdown-content">
          <Link to="/profile">
            <VscEdit /> Editar perfil
          </Link>
          <Link onClick={handleClick} to="/auth">
            <CgLogOut />
            Cerrar sesi√≥n
          </Link>
        </div>
      </div>
    );
  };

  return (
    <Fragment>
      <div className="navbar">
        <section className="navbar__logo">
          <a href="/home">Reddet</a>
        </section>
        <section className="navbar__profile">
          <Menu />
        </section>
      </div>
    </Fragment>
  );
};
const mapStateToProps = (state) => {
  return {
    auth: state,
  };
};
export default connect(mapStateToProps, { startLogoutUser, startGetUser })(
  Navbar
);
