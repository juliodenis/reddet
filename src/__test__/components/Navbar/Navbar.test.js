import React from "react";
import { shallow } from "enzyme";
import { create } from "react-test-renderer";
import ProviderMock from "../../../__mocks__/ProviderMock";
import Navbar from "../../../components/Navbar/Navbar";

describe("<Navbar/>", () => {
  test("Should render the Navbar component", () => {
    const navbar = shallow(
      <ProviderMock>
        <Navbar />
      </ProviderMock>
    );
    expect(navbar.length).toEqual(1);
  });
});

describe("Navbar Snapshot", () => {
  test("Check Navbar Snapshot", () => {
    const navbar = create(
      <ProviderMock>
        <Navbar />
      </ProviderMock>
    );
    expect(navbar.toJSON()).toMatchSnapshot();
  });
});
