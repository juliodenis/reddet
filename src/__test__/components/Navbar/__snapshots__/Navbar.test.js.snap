// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Snapshot Check Navbar Snapshot 1`] = `
<div
  className="navbar"
>
  <section
    className="navbar__logo"
  >
    <a
      href="/home"
      onClick={[Function]}
    >
      Reddet
    </a>
  </section>
  <section
    className="navbar__profile"
  >
    <div
      className="navbar__profile--settings"
    >
      <a
        className="navbar__profile--settings-dropdown"
      >
        <img
          alt="user profile"
          src={Object {}}
        />
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </a>
      <div
        className="profile-dropdown-content"
      >
        <a
          href="/profile"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.23 1h-1.46L3.52 9.25l-.16.22L1 13.59 2.41 15l4.12-2.36.22-.16L15 4.23V2.77L13.23 1zM2.41 13.59l1.51-3 1.45 1.45-2.96 1.55zm3.83-2.06L4.47 9.76l8-8 1.77 1.77-8 8z"
            />
          </svg>
           Editar perfil
        </a>
        <a
          href="/auth"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.51428 20H4.51428C3.40971 20 2.51428 19.1046 2.51428 18V6C2.51428 4.89543 3.40971 4 4.51428 4H8.51428V6H4.51428V18H8.51428V20Z"
              fill="currentColor"
            />
            <path
              d="M13.8418 17.385L15.262 15.9768L11.3428 12.0242L20.4857 12.0242C21.038 12.0242 21.4857 11.5765 21.4857 11.0242C21.4857 10.4719 21.038 10.0242 20.4857 10.0242L11.3236 10.0242L15.304 6.0774L13.8958 4.6572L7.5049 10.9941L13.8418 17.385Z"
              fill="currentColor"
            />
          </svg>
          Cerrar sesi√≥n
        </a>
      </div>
    </div>
  </section>
</div>
`;
